### PR TITLE
feat: add Restriction Type in EditCoursePage

### PR DESCRIPTION
**Pull request by zawan-ila**
[PROD-4008](https://2u-internal.atlassian.net/browse/PROD-4008)
-------
This PR adds restriction_type field to CourseRunForm and makes it visible only for external courses

Screenshot 2024-04-22 at 10 33 54 PM

## Testing

1. Check out https://github.com/openedx/course-discovery/pull/4331
2. Open an executive education course in publisher
3. Update its restriction_type field
4. Verify that the restriction_type is appropriately updated in the backend. This can be verified through the CourseRunRestriction model in Django admin

### CODE REVIEW

**Review by Ali-D-Akbar — APPROVED** [review]

File: src/utils/index.js
Comment (DawoudSheraz) [inline src/utils/index.js:374]: nit: the empty value should be first, no?

File: src/components/EditCoursePage/EditCoursePage.test.jsx
Comment (DawoudSheraz) [inline src/components/EditCoursePage/EditCoursePage.test.jsx:68]: do we need this value across all test data? Shouldn't coalesce handle the missing data?

Comment (zawan-ila) [reply src/components/EditCoursePage/EditCoursePage.test.jsx:68]: This blob of data is a "mock" for the network response. Since the network response only has `null`, `custom-b2c` and `custom-b2b-enterprise` as possible values, I am testing here with both a null and string response. This is the string version and the course run below is the null version.

File: src/utils/index.js
Comment (zawan-ila) [reply src/utils/index.js:374]: I'll update it before merging

File: src/components/EditCoursePage/EditCoursePage.test.jsx
Comment (DawoudSheraz) [reply src/components/EditCoursePage/EditCoursePage.test.jsx:68]: Got it.

**Review by DawoudSheraz — APPROVED** [review]
One small comment, otherwise looks good. Make sure to merge this after backend PR has been merged.

File: src/components/EditCoursePage/CollapsibleCourseRun.test.jsx
Comment (zawan-ila) [reply src/components/EditCoursePage/CollapsibleCourseRun.test.jsx:168]: I don't see how we can do that here. In particular, I am unable to find the select component here. I think it is because of the shallow render. I tried mounting it, but then it complained about not being inside a reduxForm.

I have added a test for the actual value in `EditCoursePage.test.jsx`.

**Pull request merged**
